File: daily/Binary Tree/111-Minimum-depth-of-binary-tree.py
class Solution:
    def minDepth(self, root: Optional[TreeNode]) -> int:
        def dfs(node):
            if not node:
                return 0
            left = dfs(node.left)
            right = dfs(node.right)

            if not node.left:
                return 1 + right
            if not node.right:
                return 1 + left

            return 1 + min(left, right)
        return dfs(root)

File: daily/Binary Tree/111-Minimum-depth-of-binary-tree.go
package daily

func minDepth(root *TreeNode) int {
	return dfsMinDepth(root)
}

func dfsMinDepth(node *TreeNode) int {
	if node == nil {
		return 0
	}

	left := dfsMinDepth(node.Left)
	right := dfsMinDepth(node.Right)
	if node.Left == nil {
		return 1 + right
	}
	if node.Right == nil {
		return 1 + left
	}
	return 1 + min(left, right)
}
